director/projects/api/views/projects.py

```diff
@@ -163,10 +150,6 @@ def retrieve(self, request: Request, pk: int) -> Response:
         Returns details of project, including `name` and `description`.
         """
         # Check that the user has VIEW permissions for the project
-        self.get_project(request.user, pk=pk)
-        if not self.has_permission(ProjectPermissionType.VIEW):
-            raise PermissionDenied
-
-        project = get_object_or_404(Project, pk=pk)
+        project = self.get_project(request.user, pk=pk)
```

Comment:
`self.get_project` takes care of raising a 404 and checking of permissions (on `self. project_permission_required`)